hs_core/migrations/0024_custom_migration_metadata_namespace_20160527.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations
+
+from hs_core.models import BaseResource
+from hs_core.hydroshare.utils import resource_modified
+
+def migrate_namespace_for_source_and_relation(apps, schema_editor):
+    # migrate the namespace for the 'Source' and 'Relation' metadata
+    # elements from 'dcterms' to 'hsterms' by regenerating the metadata xml
+    log = logging.getLogger()
+    for res in BaseResource.objects.all():
+        # need to regenerate xml file only for those resources that have either 'source' or 'relation' metadata
+        if len(res.metadata.sources.all()) > 0 or len(res.metadata.relations.all()) > 0:
+            resource_modified(res, res.creator)
+            log_msg = "Namespace for either 'source' or 'relation' metadata element was updated for resource " \
```

Comment:
@pkdash For this specific case, table schema in database has no change, right? Only data migration is needed to update resourcemetadata.xml file for relevant resources on iRODS side, right? Every time when this custom migration runs, resource_modified() will set modified date to current time as well as update resourcemetadata.xml, resourcemap.xml, and set bag_modified to true. Agreed this custom migration will not do much harm by repeated execution, but it is still not so good to update files on iRODS side when not needed. a +1 from me on this PR, though I'll defer to you on whether there is more work to be done to make this custom migration more robust.